blockchain_connector/Dockerfile

```diff
@@ -120,53 +117,157 @@ RUN mkdir -p build \
   && cmake .. -DUNTRUSTED_ONLY=1 \
   && make
 
-#install avalon common module
+
+# -------------=== common/python build ===-------------
+
+#Build common/python intermediate docker image
+FROM python_image as common_python_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    make
+
+# Copy openssl build artifacts from openssl_image
+COPY --from=openssl_image /usr/local/ssl /usr/local/ssl
+COPY --from=openssl_image /usr/local/bin /usr/local/bin
+COPY --from=openssl_image /usr/local/include /usr/local/include
+COPY --from=openssl_image /usr/local/lib /usr/local/lib
+
+RUN ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+
+ENV TCF_HOME=/project/avalon
+
+COPY ./common/python /project/avalon/common/python
+
 WORKDIR /project/avalon/common/python
 
 RUN echo "Building Avalon Common Python\n" \
-  && python setup.py bdist_wheel \
-  && echo "Installing Avalon Common Python" \
-  && pip3 install dist/*.whl
+ && make
+
+
+# -------------=== common/crypto_utils build ===-------------
+
+#Build common/cpp intermediate docker image
+FROM python_image as common_crypto_utils_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    swig \
+    make
+
+
+# Copy openssl build artifacts from openssl_image
+COPY --from=openssl_image /usr/local/ssl /usr/local/ssl
+COPY --from=openssl_image /usr/local/bin /usr/local/bin
+COPY --from=openssl_image /usr/local/include /usr/local/include
+COPY --from=openssl_image /usr/local/lib /usr/local/lib
+
+RUN ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/
+
+COPY --from=common_cpp_image /project/avalon/common/cpp/build/libuavalon-base64.a /project/avalon/common/cpp/build/
+COPY --from=common_cpp_image /project/avalon/common/cpp/build/libuavalon-common.a /project/avalon/common/cpp/build/
+COPY --from=common_cpp_image /project/avalon/common/cpp/build/libuavalon-crypto.a /project/avalon/common/cpp/build/
+COPY --from=common_cpp_image /project/avalon/common/cpp/build/libuavalon-parson.a /project/avalon/common/cpp/build/
```

Comment:
same comment as in https://github.com/hyperledger/avalon/pull/329/files#r394100498